File: CPP/weekly-contest-204/No1.cpp
/**
 * Created by Xiaozhong on 2020/8/30.
 * Copyright (c) 2020/8/30 Xiaozhong. All rights reserved.
 */
#include "vector"
#include "iostream"

using namespace std;

class Solution {
public:
    bool containsPattern(vector<int> &arr, int m, int k) {
        for (int i = 0; i < arr.size() - m; i++) {
            int count = 0;
            int j = i;
            for (int n = 0; n <= k; n++) {
                if (arr[j] == arr[m + j]) {
                    j++;
                    if (++count >= k) return true;
                    if (j - i >= m) continue;
                } else break;
            }
        }
        return false;
    }
};

int main() {
    Solution s;
    vector<int> arr = {1, 2, 4, 4, 4, 4};
    cout << s.containsPattern(arr, 1, 3) << endl;
    arr = {1, 2, 1, 2, 1, 1, 1, 3};
    cout << s.containsPattern(arr, 2, 2) << endl;
    arr = {1, 2, 1, 2, 1, 3};
    cout << s.containsPattern(arr, 2, 3) << endl;
}

File: Java/src/weeklyContest231/No1.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

package weeklyContest231;

import java.util.List;

public class No1 {
    public boolean checkOnesSegment(String s) {
        String[] list = s.split("0");
        int count = 0;
        for (String item : list) {
            if (!item.equals("")) count++;
        }
        return count <= 1;
    }

    public static void main(String[] args) {
        No1 no1 = new No1();
        System.out.println(no1.checkOnesSegment("1001"));
        System.out.println(no1.checkOnesSegment("110"));
        System.out.println(no1.checkOnesSegment("1100111"));
        System.out.println(no1.checkOnesSegment("1"));
        System.out.println(no1.checkOnesSegment("10"));
    }
}
